pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ThresholdShedder.java

```diff
@@ -233,5 +237,23 @@ private Pair<Boolean, String> getMaxUsageBroker(
         }
         return Pair.of(hasBrokerBelowLowerBound, maxUsageBrokerName);
     }
+    @Override
+    public void onBrokerChange(Set<String> newBrokers) {
+        synchronized (activeBrokers) {
+            activeBrokers.clear();
+            activeBrokers.addAll(newBrokers);
+        }
+    }
+
+    private void cleanUnActiveBroker() {
+        if (!activeBrokers.isEmpty()) {
+            synchronized (activeBrokers) {
+                if (!activeBrokers.isEmpty()) {
+                    brokerAvgResourceUsage.keySet().removeIf((key) -> !activeBrokers.contains(key));
+                    activeBrokers.clear();
```

Comment:
Why clear `activeBrokers` ?

Reply:
We don't want to clean up the inactive Broker every time, because it needs to be locked. Therefore, we need an indicator to determine whether to clean up